### ISSUE TITLE
"Docker Desktop is stopped" and unable to Quit

### ISSUE THREAD

**Issue by jgresham** [docker/for-mac#6606]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:
64C08BF7-5F22-4B05-9ED0-8E31DE0C3A20/20221206172238
64C08BF7-5F22-4B05-9ED0-8E31DE0C3A20/20221206155232
sometimes I am unable to run the diagnositcs

### Expected behavior
The app will at least allow me to quit and restart if it stops working.

### Actual behavior
The app is stuck (Docker not will not quit and says it is stopped) and "Quit Docker Desktop" from the menu or `Command-Q` doesn't quit the app so that I can restart it.

### Information

This has happened to me before. It usually happens when I open my macbook lid, after closing my macbook lid while docker is running beforehand.
  - macOS Version: 12.6
  - Intel chip or Apple chip: Apple M2
  - Docker Desktop Version: 4.15.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
```
/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[FAIL] DD0001: is the application running? Docker is not running
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[FAIL] DD0011: are the LinuxKit services running? failed to ping VM diagnosticsd with error: Get "http://ipc/ping": dial unix diagnosticd.sock: connect: no such file or directory
[2022-12-02T01:02:25.210252000Z][com.docker.diagnose][I] ipc.NewClient: 199958a5-diagnose -> diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.glob..func12()
[	common/pkg/diagkit/gather/diagnose/linuxkit.go:18 +0x8c
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x105933960)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x40
[common/pkg/diagkit/gather/diagnose.Run.func1(0x105933960)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x40
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x4?, 0x105933960)
[	common/pkg/diagkit/gather/diagnose/run.go:142 +0x80
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x3, 0x105933960, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:151 +0x88
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x2, 0x1059339e0, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x50
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x105933ae0, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x50
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x105933d60, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x50
[common/pkg/diagkit/gather/diagnose.walkOnce(0x10540c700?, 0x140003df890)
[	common/pkg/diagkit/gather/diagnose/run.go:137 +0x8c
[common/pkg/diagkit/gather/diagnose.Run(0x105933d60, 0x2edaf308?, {0x140003dfb08, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x160
[main.checkCmd({0x140001b4010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:133 +0xdc
[main.main()
[	common/cmd/com.docker.diagnose/main.go:99 +0x30c
[2022-12-02T01:02:25.210923000Z][com.docker.diagnose][I] (32e9a2c1) 199958a5-diagnose C->S diagnosticsd GET /ping
[2022-12-02T01:02:25.212629000Z][com.docker.diagnose][W] (32e9a2c1) 199958a5-diagnose C<-S NoResponse GET /ping (1.68275ms): Get "http://ipc/ping": dial unix diagnosticd.sock: connect: no such file or directory

[FAIL] DD0004: is the Docker engine running? Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:25.213497000Z][com.docker.diagnose][I] ipc.NewClient: ce14191b-com.docker.diagnose -> lifecycle-server.sock VMDockerdAPI
[linuxkit/pkg/desktop-host-tools/pkg/client.NewClientForPath(...)
[	linuxkit/pkg/desktop-host-tools/pkg/client/client.go:61
[linuxkit/pkg/desktop-host-tools/pkg/client.NewClient({0x105241c45, 0x13})
[	linuxkit/pkg/desktop-host-tools/pkg/client/client.go:55 +0x90
[common/pkg/diagkit/gather/diagnose.isDockerEngineRunning()
[	common/pkg/diagkit/gather/diagnose/dockerd.go:21 +0x28
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x1059339e0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x40
[common/pkg/diagkit/gather/diagnose.Run.func1(0x1059339e0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x40
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x3?, 0x1059339e0)
[	common/pkg/diagkit/gather/diagnose/run.go:142 +0x80
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x2, 0x1059339e0, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:151 +0x88
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x105933ae0, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x50
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x105933d60, 0x14000607718)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x50
[common/pkg/diagkit/gather/diagnose.walkOnce(0x10540c700?, 0x140003df890)
[	common/pkg/diagkit/gather/diagnose/run.go:137 +0x8c
[common/pkg/diagkit/gather/diagnose.Run(0x105933d60, 0x2edaf308?, {0x140003dfb08, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x160
[main.checkCmd({0x140001b4010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:133 +0xdc
[main.main()
[	common/cmd/com.docker.diagnose/main.go:99 +0x30c
[2022-12-02T01:02:25.213990000Z][com.docker.diagnose][I] (5144c82b) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /docker
[2022-12-02T01:02:25.214221000Z][com.docker.diagnose][W] (5144c82b) ce14191b-com.docker.diagnose C<-S NoResponse GET /docker (229.916µs): Get "http://ipc/docker": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:25.214420000Z][com.docker.diagnose][I] (5144c82b-1) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:25.214645000Z][com.docker.diagnose][W] (5144c82b-1) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (223.375µs): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:26.215193000Z][com.docker.diagnose][I] (5144c82b-2) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:26.216356000Z][com.docker.diagnose][W] (5144c82b-2) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (1.160291ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:27.217258000Z][com.docker.diagnose][I] (5144c82b-3) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:27.219943000Z][com.docker.diagnose][W] (5144c82b-3) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (2.685583ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:28.221341000Z][com.docker.diagnose][I] (5144c82b-4) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:28.223269000Z][com.docker.diagnose][W] (5144c82b-4) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (1.949167ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:29.224363000Z][com.docker.diagnose][I] (5144c82b-5) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:29.228220000Z][com.docker.diagnose][W] (5144c82b-5) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (3.84425ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:30.231184000Z][com.docker.diagnose][I] (5144c82b-6) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:30.233011000Z][com.docker.diagnose][W] (5144c82b-6) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (1.845459ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:31.234704000Z][com.docker.diagnose][I] (5144c82b-7) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:31.235708000Z][com.docker.diagnose][W] (5144c82b-7) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (1.003917ms): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory
[2022-12-02T01:02:32.236452000Z][com.docker.diagnose][I] (5144c82b-8) ce14191b-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-12-02T01:02:32.237440000Z][com.docker.diagnose][W] (5144c82b-8) ce14191b-com.docker.diagnose C<-S NoResponse GET /ping (983.625µs): Get "http://ipc/ping": dial unix lifecycle-server.sock: connect: no such file or directory

[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? Cannot connect to the Docker daemon at unix://docker.raw.sock. Is the docker daemon running?
[PASS] DD0013: is the $PATH ok?
```
### Steps to reproduce the behavior

  1. Just came back to my Macbook and saw this. Not sure how it happened.

**Comment by jgresham** [comment]
After force quitting Docker Desktop in the activity monitor app, I cannot start Docker Desktop. Only the following error is shown. Resetting settings doesn't fix it.

<img width="595" alt="Captura de Pantalla 2022-12-01 a la(s) 5 24 46 p m" src="https://user-images.githubusercontent.com/3721291/205193846-3afc24ec-9791-4b91-9da7-b2af9dcbdb5c.png">

**Comment by jgresham** [comment]
After force quitting 2 other com.docker.backend services, I can now start Docker

**Comment by ackerthehacker2** [comment]
I have seen this on Ventura too ...  going back tp 4.12 and it seems to most stable.    

Diag.  5FF80EFF-E372-4E3E-9C2E-C79FEF30BE88/20221205002705

**Comment by jgresham** [comment]
Other's I know are having this issue. And I keep getting this issue intermittently on my Macbook.

Some Diagnostic IDs reported are:
64C08BF7-5F22-4B05-9ED0-8E31DE0C3A20/20221206172238
64C08BF7-5F22-4B05-9ED0-8E31DE0C3A20/20221206155232

**Comment by gman-wa** [comment]
I am having this same issue.  Reverted to 4.14.1

**Comment by sattellite** [comment]
Same issue. Diag `ED00D8A5-3697-4015-A03B-F0047FABFB15/20221208094055`

**Comment by jgresham** [comment]
This issue seems related https://github.com/docker/for-mac/issues/6507

**Comment by jgresham** [comment]
@nicks sorry for the tag, but just wanted to draw attention to this as it is a critical issue for many. Curious if the team is aware or working on a fix for this? 

Best,
Johns

**Comment by nicks** [comment]
@jgresham I poked around a bit in your diagnostics and saw some vpnkit segfaults.

If it crashes again, can you check under ~/Library/Logs/DiagnosticReports and upload anything that says com.docker.vpnkit? 

My hunch is that this is likely the same issue as https://github.com/docker/for-mac/issues/6530. (a crashdump will confirm for sure).

I think the current working theory is that there's a low-level sequence of packets that makes all the host/docker network bridging blow up, but we're still trying to narrow it down. isolated repro cases help. (The users in that thread have done a good job getting something repro-able but it's still a pretty "large" repro case.)

**Comment by nicks** [comment]
also i usually do a `pkill docker` to kill the whole tree of docker processes; killing from the activity monitor is likely to leave some orphans.

**Comment by Eleleth77** [comment]
Hey! First post.. hope this works
[DiagnosticReport.zip](https://github.com/docker/for-mac/files/10191200/DiagnosticReport.zip)

**Comment by gman-wa** [comment]
Here are my vpnkit DiagnoticReports
[DiagnosticReports.zip](https://github.com/docker/for-mac/files/10197377/DiagnosticReports.zip)

**Comment by 11ib** [comment]
+1 on this, here's my DiagnosticReport:

[DiagnosticReport.zip](https://github.com/docker/for-mac/files/10198897/DiagnosticReport.zip)

**Comment by jgresham** [comment]
One more
[com.docker.vpnkit-2022-12-09-162845.ips.zip](https://github.com/docker/for-mac/files/10199175/com.docker.vpnkit-2022-12-09-162845.ips.zip)

**Comment by jgresham** [comment]
Two more 
[Archive.zip](https://github.com/docker/for-mac/files/10241359/Archive.zip)

**Comment by Eleleth77** [comment]
just updated to the latest version of Docker which looked like it addressed most the mac issues.. still crashed within first few minutes

[com.docker.vpnkit-2022-12-19-231322.ips.zip](https://github.com/docker/for-mac/files/10265229/com.docker.vpnkit-2022-12-19-231322.ips.zip)

**Comment by nicks** [comment]
we posted a debug build here that we think should resolve this problem: https://github.com/docker/for-mac/issues/6530#issuecomment-1344029769

(looking at @jgresham 's stack trace, the bug is slightly different, but i think the same fix will work)

**Comment by Eleleth77** [comment]
> we posted a debug build here that we think should resolve this problem: [#6530 (comment)](https://github.com/docker/for-mac/issues/6530#issuecomment-1344029769)
> 
> (looking at @jgresham 's stack trace, the bug is slightly different, but i think the same fix will work)

3 hours strong so far.. looking good, will check back in the morning. Really appreciate the work 🙏🏽

**Comment by Eleleth77** [comment]
> we posted a debug build here that we think should resolve this problem: [#6530 (comment)](https://github.com/docker/for-mac/issues/6530#issuecomment-1344029769)
> 
> (looking at @jgresham 's stack trace, the bug is slightly different, but i think the same fix will work)

no dice. Crashed after 4+ hours though

Report ID
64C08BF7-5F22-4B05-9ED0-8E31DE0C3A20/20221221041902
[com.docker.vpnkit-2022-12-19-231322.ips.zip](https://github.com/docker/for-mac/files/10274543/com.docker.vpnkit-2022-12-19-231322.ips.zip)

**Comment by oenu** [comment]
Similar issues, really feel like the Mac is not a concern for the docker team at this point

**Comment by stratacorps** [comment]
Tried 4.15.0 today...lasted about an hour before it stopped. As others have posted, 4.12.0 doesn't do this...I have tried all releases after that which fail.

**Comment by HawaiiRyan** [comment]
Same as everyone else w/ Intel macOS. 4.12 is solid

**Comment by ackerthehacker2** [comment]
4,15.1 and 4.16 is pretty solid on M1 Mac Studio …. I’ve been getting 2-3 days of running before dicker stops …better than what I got with 4.12

**Comment by gman-wa** [comment]
> 4,15.1 and 4.16 is pretty solid on M1 Mac Studio …. I’ve been getting 2-3 days of running before dicker stops …better than what I got with 4.12

I think for most of us that does not qualify as 'pretty solid'.  Pretty 'the other thing' probably.  :)

**Comment by jgresham** [comment]
@ackerthehacker2 @gman-wa unfortunately, this level of crashing/performance is not good enough for me. I'm evaluating Podman as an alternative to Docker (https://podman.io). Please let me know if you try it out and if you have a good experience.

**Comment by nicks** [comment]
A fix has been released in Docker Desktop 4.16. See the [release notes](https://docs.docker.com/desktop/release-notes/) for more details.

If you have other issues with Docker Desktop stopping unexpectedly, please file a new issue with diagnostics and/or repro steps. A big reason we were able to fix the vpnkit crashes is because the discussion on https://github.com/docker/for-mac/issues/6530 was able to isolate a repro case. Thanks!

**Comment by oenu** [comment]
Issue not resolved
